pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -305,6 +305,21 @@ func (e *EndpointController) syncService(key string) {
 		e.queue.Add(key) // Retry
 		return
 	}
+	if len(pods.Items) == 0 {
+		// When controller manager starts, usually service list would be faster than pod list;
+		// So list pods from cache would get empty list;
+		// The risk is endpoints controller then will flush the endpoints for this service;
+		// If both controller manager and kube-proxy are using service VIP to access apiserver,
+		// they will fail after endpoints flush, and there is no way to automatically recover.
```

Comment:
@bprashanth What I observed is iptables keep there to dnat outgoing packets to kube-proxy, but kube-proxy can not do any forwarding.